server/conn_stmt.go

```diff
@@ -149,6 +150,10 @@ func (cc *clientConn) handleStmtExecute(ctx context.Context, data []byte) (err e
 		useCursor = false
 	case 1:
 		useCursor = true
+	case 128: // highest bit set (0x80) so unsigned, no flags (0)
+		useCursor = false
+	case 129: // highest bit set (0x80) so unsigned, flag CURSOR_TYPE_READ_ONLY set.
+		useCursor = true
```

Comment:
I don't  think it is the correct fix. Since `signed/unsigned` flag is for `enum_field_types`, here the flag is `enum_cursor_type`. We should mimic mysql:

```
if flag & CURSOR_TYPE_READ_ONLY {
  useCursor = true
}
```

Reply:
I've tried to address this in 179052cf70b038aaf6e4e067d846680bf6ea9247
PTAL @xhebox

Reply:
Not a correct definition. Say `0x01`, will `flag&mysql.CursorTypeNoCursor > 0`.

Should be something like:

```
const (
	CursorTypeReadOnly uint8 = 1 << iota
	CursorTypeForUpdate
	CursorTypeScrollable
)
```

Reply:
Fixed this in 3b89e33da7fb8975b56c172d67ddfe24ca7989bc
PTAL @xhebox